src/sinks/aws_s_s/sink.rs

```diff
@@ -39,6 +39,7 @@ where
     C: Client<E> + Clone + Send + Sync + 'static,
     E: std::fmt::Debug + std::fmt::Display + std::error::Error + Sync + Send + 'static,
 {
+    #[allow(dead_code)]
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏Was this added by mistake? It seems like it shouldn't be necessary since we're using the `new()`function.

Side note- this looks like another `pub` that could be turned into a `pub(super)`